src/ParallelAlgorithms/Amr/Actions/UseAmrDiagnostics.hpp

```diff
@@ -0,0 +1,71 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <boost/rational.hpp>
+#include <cstddef>
+#include <vector>
+
+#include "DataStructures/DataBox/DataBox.hpp"
+#include "Domain/Domain.hpp"
+#include "Domain/Tags.hpp"
+#include "Parallel/GlobalCache.hpp"
+#include "Parallel/Printf.hpp"
+#include "Parallel/Reduction.hpp"
+#include "Utilities/MakeString.hpp"
+#include "Utilities/StdHelpers.hpp"
+#include "Utilities/System/Abort.hpp"
+
+namespace amr::Actions {
+
+/// \brief Use the AMR diagnostics gathered from all of the Element%s
+///
+/// Checks the following:
+/// - That the fraction of Block volume (in the logical coordinate frame)
+/// covered by all Element%s is equal to the number of Block%s in the Domain
+///
+/// Prints the following (as integers):
+/// - The number of elements
+/// - The number of grid points
+/// - The average refinement level by logical dimension (i.e. not by the
+///   physical dimensions)
+/// - The average number of grid points by logical dimension
+struct UseAmrDiagnostics {
```

Comment:
`CheckAmrDiagnostics`? Not sure what "use" means.

Reply:
That's what I had named it when it just checked the Block volume.  But then I didn't think that was best after I started doing additional things.  And I can easily imagine more things being done as we start to use AMR.  @nilsdeppe @wthrowe thoughts on the name?

Reply:
What about `RunAmrDiagnostics`?